### PR TITLE
Add name validation to acorn update --pull to block nested acorns

### PR DESCRIPTION
**Pull request by g-linville**
context: https://github.com/acorn-io/acorn/issues/1710#issuecomment-1579387475

for: #1710

### Checklist
- [x] The title of this PR would make a good line in Acorn's Release Note's Changelog
- [x] The title of this PR ends with a link to the main issue being address in parentheses, like: `This is a title (#1216)`. [Here's an example](https://github.com/acorn-io/acorn/pull/1199)
- [x] All relevant issues are referenced in the PR description. *NOTE: don't use [GitHub keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) that auto-close issues*
- [x] Commits follow [contributing guidance](https://github.com/acorn-io/acorn/blob/main/CONTRIBUTING.md#commits)
- [ ] Automated tests added to cover the changes. If tests couldn't be added, an explanation is provided in the Verification and Testing section
- [x] Changes to user-facing functionality, API, CLI, and upgrade impacts are clearly called out in PR description
- [ ] PR has at least two approvals before merging (or a reasonable exception, like it's just a docs change)

### CODE REVIEW

File: pkg/server/registry/apigroups/acorn/apps/pull.go [hunk 1]
```diff
@@ -50,3 +56,12 @@ func (s *PullAppImageStrategy) Create(ctx context.Context, obj types.Object) (ty
 func (s *PullAppImageStrategy) New() types.Object {
 	return &apiv1.AppPullImage{}
 }
+
+func (v PullAppImageNameValidator) ValidateName(ctx context.Context, obj runtime.Object) (result field.ErrorList) {
+	appPullImage := obj.(*apiv1.AppPullImage)
+	if len(strings.Split(appPullImage.Name, ".")) == 2 {
+		result = append(result, field.Invalid(field.NewPath("metadata", "name"), appPullImage.Name, "invalid name\nTo update a nested Acorn or a service, update the parent Acorn instead."))
```
Comment (njhale) [inline pkg/server/registry/apigroups/acorn/apps/pull.go:63]: nit: I think "invalid name" is implied by the field path and error type, so we could drop it (and the new-line) from the description.

**Review by njhale — APPROVED** [review]
one nit, but nothing to block this over

File: pkg/server/registry/apigroups/acorn/apps/pull.go (hunk 1 above)
Comment (g-linville) [reply pkg/server/registry/apigroups/acorn/apps/pull.go]: :+1: updated

**Review by njhale — APPROVED** [review]
lgtm

**Review by cjellick — APPROVED** [review]

**Pull request merged**